### DISCUSSION TITLE
3 Acova Percale2 but only 2 reconized

### DISCUSSION THREAD

**Discussion by llaumgui** [Koenkk/zigbee2mqtt#18355] [Devices]
Hi,

I migrate from zigpy (plugin Zigbee on Jeedom) to Zigbee2MQTT (plugin jeeZigbee on Jeedom). On zigpy, no soucy.

I have bought 3 [Acova Percale 2](https://www.zigbee2mqtt.io/devices/PERCALE2.html). 2 of them are reconized witout any problem but the third is marked unsupported and display “TAFFETAS2 D1.00P1.03Z1.00” Instead of “PERCALE2 D1.00P1.03Z1.00” (remark: [Taffeta is also supported](https://www.zigbee2mqtt.io/devices/TAFFETAS2.html)).

Is there a way to map my Percale like a Percale ?

**Comment by Koenkk** [comment]
Added!

Changes will be available in the dev branch in a few hours from now. (https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html)

**Comment by molusk** [comment]
Hello @llaumgui,

If I understand well, you have 2 Acova Percale2 heaters which think they are Taffetas2 ?
It seems Percale2 and Taffetas2 have the exact same features (including presence and open window detection), the heating technology seems to be the only difference between them.

On the other hand, Zigbee2MQTT integration is not at the same level for both. I added a few features to Taffetas2 converter lately : motion sensor and temperature calibration. A Taffetas2 user is complaining to me my late additions don’t show up with his heaters and I think he may have Taffetas2 heaters with the same firmware as yours (I asked him to confirm that).

So, my question, what do you think about making the configuration clean ? It means, at least, putting all Taffetas2 definitions together and maybe add the features to Percale2. I can do the configuration modifications but I can’t test anything as I don’t own the impacted devices. You would have to confirm everything is working well (using a custom converter before pushing anything to github of course).

The impact for you, once it is done, is having a bit more control on your heaters (useful or not, it’s your call) but 2 of them would be identified as Taffetas2 instead of Percale2 unless you keep a custom converter for them but you have to follow future configuration evolution manually.

Anyway, I won’t do anything without your agreement as I am not impacted and you are.

**Comment by llaumgui** [comment]
Hi,

I have 3 Acova Percale2. But only one is reconized like a Taffetas2.

For me:
 - Electronic of Percale2 and Taffetas2 is identical.
 - Features of Percale2 and Taffetas2 are identical.
 - Look & feel are identical.
 - Percale2 is a fluid inertia heater and Taffetas2 is a Dry inertia heater. It's the only difference.
 - Acova can mount a Taffetas electronic on a Percale whitout issue.

For me you can clean configuration and allow same feature for Percale2 and Taffetas.

Best regards.

**Comment by molusk** [comment]
Ok, I will share a custom converter here soon and notify you so you can confirm everything is ok in your configuration.
Thanks.

**Reply by llaumgui** [reply]
@molusk Do you have some elements to test ?

**Reply by molusk** [reply]
Sorry @llaumgui I completely forget to create the custom converter...

**Reply by molusk** [reply]
[acova-custom.zip](https://github.com/Koenkk/zigbee2mqtt/files/12847439/acova-custom.zip)
@llaumgui here is a first test for the clean acova converter, it works on my side but I only have taffetas.

You should have 2 Percale and 1 Taffetas with this converter. Devices descriptions should include "- custom", local temperature calibration should be available for both Percale and Taffetas heaters in Z2M/Device/Expose tab along with occupancy sensors.

Occupancy sensors are a bit strange, they are switched on when z2m restarts and stay on for 2 hours, then they will be triggered normally **BUT** will still stay on for 2 hours. I think Acova made this as an anti-hysteresis feature, this is quite an ugly solution but we will have to live with it (forever as there is no firmware updates available...).

**Reply by llaumgui** [reply]
I looking the diff... It's look fine, I install this. Juste a remark: Why not merge PERCALE2 and TAFFETAS2 and calling that "PERCALE2 / TAFFETAS2" ? Because the modèle is juste given by the electronic module but body of the heater can be one or other ?

**Reply by llaumgui** [reply]
I trying your config: my false TAFFETAS2 is OK but I have loose command of all my PERCAL2 heater...

**Reply by llaumgui** [reply]
Hum... @molusk modelId of percal are: "PERCALE2 D1.00P1.03Z1.00\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000" not "PERCALE2 D1.00P1.03Z1.00"... But the official configuration currently working.

**Reply by llaumgui** [reply]
OK, this work fine for detection:

```javascript
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const e = exposes.presets;

const definitions = [
  {
    zigbeeModel: [
      "PERCALE2 D1.00P1.01Z1.00\u0000\u0000\u0000\u0000\u0000\u0000\u0000",
      "PERCALE2 D1.00P1.02Z1.00\u0000\u0000\u0000\u0000\u0000\u0000\u0000",
      "PERCALE2 D1.00P1.03Z1.00\u0000\u0000\u0000\u0000\u0000\u0000\u0000",
    ],
    model: "PERCALE2",
    vendor: "Acova",
    description: "Percale 2 heater - custom",
    fromZigbee: [fz.thermostat, fz.hvac_user_interface],
    toZigbee: [
      tz.thermostat_local_temperature,
      tz.thermostat_system_mode,
      tz.thermostat_occupied_heating_setpoint,
      tz.thermostat_unoccupied_heating_setpoint,
      tz.thermostat_occupied_cooling_setpoint,
      tz.thermostat_running_state,
      tz.thermostat_local_temperature_calibration,
    ],
    exposes: [
      e
        .climate()
        .withSetpoint("occupied_heating_setpoint", 7, 28, 0.5)
        .withSetpoint("unoccupied_heating_setpoint", 7, 28, 0.5)
        .withLocalTemperature()
        .withSystemMode(["off", "heat", "auto"])
        .withRunningState(["idle", "heat"])
        .withLocalTemperatureCalibration(),
      e.occupancy(),
    ],
    configure: async (device, coordinatorEndpoint, logger) => {
      const endpoint = device.getEndpoint(1);
      const endpoint2 = device.getEndpoint(2);
      await reporting.bind(endpoint, coordinatorEndpoint, [
        "genPowerCfg",
        "hvacThermostat",
      ]);
      await reporting.bind(endpoint2, coordinatorEndpoint, [
        "msOccupancySensing",
      ]);
      await reporting.thermostatTemperature(endpoint);
      await reporting.thermostatRunningState(endpoint);
      await reporting.thermostatOccupiedHeatingSetpoint(endpoint);
      await reporting.thermostatUnoccupiedHeatingSetpoint(endpoint);
      await reporting.thermostatTemperatureCalibration(endpoint);
      await reporting.occupancy(endpoint2);
    },
  },
  {
    zigbeeModel: [
      "ALCANTARA2 D1.00P1.01Z1.00\u0000\u0000\u0000\u0000\u0000\u0000",
      "ALCANTARA2 D1.00P1.02Z1.00\u0000\u0000\u0000\u0000\u0000\u0000",
    ],
    model: "ALCANTARA2",
    vendor: "Acova",
    description: "Alcantara 2 heater - custom",
    fromZigbee: [fz.thermostat, fz.hvac_user_interface],
    toZigbee: [
      tz.thermostat_local_temperature,
      tz.thermostat_system_mode,
      tz.thermostat_occupied_heating_setpoint,
      tz.thermostat_unoccupied_heating_setpoint,
      tz.thermostat_running_state,
    ],
    exposes: [
      e
        .climate()
        .withSetpoint("occupied_heating_setpoint", 7, 28, 0.5)
        .withSetpoint("unoccupied_heating_setpoint", 7, 28, 0.5)
        .withLocalTemperature()
        .withSystemMode(["off", "heat", "auto"])
        .withRunningState(["idle", "heat"]),
    ],
    configure: async (device, coordinatorEndpoint, logger) => {
      const endpoint = device.getEndpoint(1);
      await reporting.bind(endpoint, coordinatorEndpoint, [
        "genPowerCfg",
        "hvacThermostat",
      ]);
      await reporting.thermostatTemperature(endpoint);
      await reporting.thermostatRunningState(endpoint);
      await reporting.thermostatOccupiedHeatingSetpoint(endpoint);
      await reporting.thermostatUnoccupiedHeatingSetpoint(endpoint);
    },
  },
  {
    zigbeeModel: [
      "TAFFETAS2 D1.00P1.02Z1.00\u0000\u0000\u0000\u0000\u0000\u0000\u0000",
      "TAFFETAS2 D1.00P1.01Z1.00\u0000\u0000\u0000\u0000\u0000\u0000\u0000",
      "TAFFETAS2 D1.00P1.03Z1.00\u0000\u0000\u0000\u0000\u0000\u0000\u0000",
    ],
    model: "TAFFETAS2",
    vendor: "Acova",
    description: "Taffetas 2 heater - custom",
    fromZigbee: [fz.thermostat, fz.hvac_user_interface, fz.occupancy],
    toZigbee: [
      tz.thermostat_local_temperature,
      tz.thermostat_system_mode,
      tz.thermostat_occupied_heating_setpoint,
      tz.thermostat_unoccupied_heating_setpoint,
      tz.thermostat_occupied_cooling_setpoint,
      tz.thermostat_running_state,
      tz.thermostat_local_temperature_calibration,
    ],
    exposes: [
      e
        .climate()
        .withSetpoint("occupied_heating_setpoint", 7, 28, 0.5)
        .withSetpoint("unoccupied_heating_setpoint", 7, 28, 0.5)
        .withLocalTemperature()
        .withSystemMode(["off", "heat", "auto"])
        .withRunningState(["idle", "heat"])
        .withLocalTemperatureCalibration(),
      e.occupancy(),
    ],
    configure: async (device, coordinatorEndpoint, logger) => {
      const endpoint = device.getEndpoint(1);
      const endpoint2 = device.getEndpoint(2);
      await reporting.bind(endpoint, coordinatorEndpoint, [
        "genPowerCfg",
        "hvacThermostat",
      ]);
      await reporting.bind(endpoint2, coordinatorEndpoint, [
        "msOccupancySensing",
      ]);
      await reporting.thermostatTemperature(endpoint);
      await reporting.thermostatRunningState(endpoint);
      await reporting.thermostatOccupiedHeatingSetpoint(endpoint);
      await reporting.thermostatUnoccupiedHeatingSetpoint(endpoint);
      await reporting.thermostatTemperatureCalibration(endpoint);
      await reporting.occupancy(endpoint2);
    },
  },
];

module.exports = definitions;
```

But only the Taffta have calibration. An idea @molusk ?

**Reply by molusk** [reply]
I don’t understand the issue with the trailing strange characters in the `modelId`. Some say they don’t need them, here I always needed them.
In your case, it seems you only need them when using a custom converter.. maybe @Koenkk has an idea why ?

Regarding your idea of fusion of the two different heaters, that’s a good idea if we can be sure they completely work the same way. The issue with the local temperature calibration may show they are not the same...
As Taffetas works on your side, maybe you can test the fusion in your custom converter : copy Percale `modelIds` in Taffetas device definition, delete Percale definition and rename Taffetas model to include Percale.
Then if Percale heaters work correctly, including temperature calibration, I think we can merge the configuration (and it means I someway messed up something in the Percale section of the custom converter).

But before doing all this work, you can check temperature calibration is available on your Percale heaters via Z2M dev console :
Open the device in Z2M, switch to dev console tab, select endpoint `1` and feature group `hvacThermostat` then select `localTemperatureCalibration` attribute and click `Read` green button below. If you get an error, your heater does not have this feature...

Here is an example with one of my Taffetas heaters.

![image](https://github.com/Koenkk/zigbee2mqtt/assets/17114364/e6cf0c4d-88f7-4f77-83c0-21ba3203896b)

**Reply by llaumgui** [reply]
Same for me :

```
 2023-10-22 09:58:46Read result of 'hvacThermostat': {"localTemperatureCalibration":0}
```

**Reply by llaumgui** [reply]
OK, I have find the issue. Taffta have "custom" in description but no PERCALE... I thinks there is a mistake in the Id detection.

**Reply by molusk** [reply]
Ok, can you check in database.db file what Z2M finds as `modelId` ?
Here is an example. (database.db has it on one line, I used a beautifier to make it human readable)
```json
{
  "id": 2,
  "type": "Router",
  "ieeeAddr": "0x5c0272fffed20e26",
  "nwkAddr": 18274,
  "manufId": 4098,
  "manufName": "ZEHNDER GROUP VAUX ANDIGNY      ",
  "powerSource": "Mains (single phase)",
  "modelId": "TAFFETAS2 D1.00P1.02Z1.00\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000\\u0000",
  "epList": [
    1,
    2,
    3,
    242
  ],
  "endpoints": {
    "1": {
      "profId": 260,
      "epId": 1,
      "devId": 769,
      "inClusterList": [
        0,
        3,
        513
      ],
      "outClusterList": [
        3
      ],
      "clusters": {
        "genBasic": {
          "attributes": {
            "hwVersion": 0,
            "zclVersion": 3
          }
        },
        "hvacThermostat": {
          "attributes": {
            "localTemp": 700,
            "runningState": 0,
            "occupiedHeatingSetpoint": 1900,
            "unoccupiedHeatingSetpoint": 700,
            "systemMode": 0,
            "ocupancy": 1,
            "localTemperatureCalibration": 0,
            "ctrlSeqeOfOper": 2,
            "runningMode": 0,
            "setpointChangeSource": 0
          }
        },
        "hvacUserInterfaceCfg": {
          "attributes": {}
        }
      },
      "binds": [
        {
          "cluster": 1,
          "type": "endpoint",
          "deviceIeeeAddress": "0x00158d0005d2a069",
          "endpointID": 1
        },
        {
          "cluster": 513,
          "type": "endpoint",
          "deviceIeeeAddress": "0x00158d0005d2a069",
          "endpointID": 1
        }
      ],
      "configuredReportings": [
        {
          "cluster": 513,
          "attrId": 0,
          "minRepIntval": 0,
          "maxRepIntval": 3600,
          "repChange": 10
        },
        {
          "cluster": 513,
          "attrId": 41,
          "minRepIntval": 0,
          "maxRepIntval": 3600,
          "repChange": 0
        },
        {
          "cluster": 513,
          "attrId": 18,
          "minRepIntval": 0,
          "maxRepIntval": 3600,
          "repChange": 10
        },
        {
          "cluster": 513,
          "attrId": 20,
          "minRepIntval": 0,
          "maxRepIntval": 3600,
          "repChange": 10
        },
        {
          "cluster": 513,
          "attrId": 16,
          "minRepIntval": 0,
          "maxRepIntval": 3600,
          "repChange": 0
        }
      ],
      "meta": {}
    },
    "2": {
      "profId": 260,
      "epId": 2,
      "devId": 263,
      "inClusterList": [
        0,
        3,
        1030
      ],
      "outClusterList": [
        3
      ],
      "clusters": {
        "genBasic": {
          "attributes": {
            "hwVersion": 0
          }
        },
        "msOccupancySensing": {
          "attributes": {
            "occupancy": 1,
            "occupancySensorType": 0
          }
        }
      },
      "binds": [
        {
          "cluster": 1030,
          "type": "endpoint",
          "deviceIeeeAddress": "0x00158d0005d2a069",
          "endpointID": 1
        }
      ],
      "configuredReportings": [
        {
          "cluster": 1030,
          "attrId": 0,
          "minRepIntval": 0,
          "maxRepIntval": 3600,
          "repChange": 0
        }
      ],
      "meta": {}
    },
    "3": {
      "profId": 260,
      "epId": 3,
      "devId": 1,
      "inClusterList": [
        0,
        3,
        15
      ],
      "outClusterList": [
        3
      ],
      "clusters": {
        "genBasic": {
          "attributes": {
            "hwVersion": 0
          }
        },
        "genBinaryInput": {
          "attributes": {
            "presentValue": 0,
            "statusFlags": 0
          }
        }
      },
      "binds": [
        {
          "cluster": 15,
          "type": "endpoint",
          "deviceIeeeAddress": "0x00158d0005d2a069",
          "endpointID": 1
        }
      ],
      "configuredReportings": [
        {
          "cluster": 15,
          "attrId": 85,
          "minRepIntval": "0",
          "maxRepIntval": 3600,
          "repChange": 0
        }
      ],
      "meta": {}
    },
    "242": {
      "profId": 41440,
      "epId": 242,
      "devId": 97,
      "inClusterList": [],
      "outClusterList": [
        33
      ],
      "clusters": {},
      "binds": [],
      "configuredReportings": [],
      "meta": {}
    }
  },
  "hwVersion": 0,
  "dateCode": "20200710D1FH1X3\\u0000",
  "zclVersion": 3,
  "interviewCompleted": true,
  "meta": {
    "configured": -1391582193
  },
  "lastSeen": 1698046998592,
  "defaultSendRequestWhen": "immediate"
}
```

**Reply by llaumgui** [reply]
I check that, but for me, there is another error. 
I thought that localTemperatureCalibration allow to change the room temperature. But it's seem to be a factor (-3° to +6°) from the local temperature... But the local temperature is not the temperature of the room but the temperature of the set point.

Look your example:

~~~json
        "hvacThermostat": {
          "attributes": {
            "localTemp": 700,
            "runningState": 0,
            "occupiedHeatingSetpoint": 1900,
            "unoccupiedHeatingSetpoint": 700,
            "systemMode": 0,
            "ocupancy": 1,
            "localTemperatureCalibration": 0,
            "ctrlSeqeOfOper": 2,
            "runningMode": 0,
            "setpointChangeSource": 0
          }
        },
~~~

Temperature of your room is it 7°00 ? Temperature on my heater is currently 23°3 (read on screen), but temperature of my sonde is 23,6. How to fix it ?

**Reply by molusk** [reply]
How did you get -3 and +6° limits ? Z2M UI allows me to set a value between -12.8 and +12.7°.
I did not test temperature calibration yet, I play with my heater management only when I don’t need them 😄 so I added the feature at the end of the cold season. It’s a new feature for me too.

One of the many limitations of Acova heaters is we can’t get the temperature sensor value... Local temperature is actually the current set-point (in my case, frost protection mode (7°) is active, shown as `heat` in Z2M, another limitation...).

My plan is to test the local temperature calibration early in the cold season with my kids room were temperature sensor is beside the bed while the heater in quite far from there so I have about 0.5° difference between them. I hope it will make the heater push to 19.5° when I ask 19° but it kind of a leap of faith 🤣 

Another issue, maybe, is the temperature calibration seems to have no effects on any part of the heater interface (neither the set-point, nor the sensor) so we can’t be sure it is effective...

If you live in a colder region than mine, you may test all of these earlier than me 😉

**Reply by llaumgui** [reply]
> How did you get -3 and +6° limits ? Z2M UI allows me to set a value between -12.8 and +12.7°.

My fault, I write an mistake.

> My plan is to test the local temperature calibration early in the cold season with my kids room were temperature sensor is beside the bed while the heater in quite far from there so I have about 0.5° difference between them. I hope it will make the heater push to 19.5° when I ask 19° but it kind of a leap of faith 🤣

For that, you can increase the `localTemp`.

> If you live in a colder region than mine, you may test all of these earlier than me 😉

I live in the south of the France... Currently, all my heaters are off.

> Another issue, maybe, is the temperature calibration seems to have no effects on any part of the heater interface (neither the set-point, nor the sensor) so we can’t be sure it is effective...

If you look to this video (and speak french), you can see that you can associate your heater to a temperature sensor : https://www.youtube.com/watch?v=gXSuO8J4Tvk (~2mn35) I thinks it's for calibration.

**Reply by molusk** [reply]
> For that, you can increase the `localTemp`.

Unless temperature calibration works... we will know soon. 

> I live in the south of the France... Currently, all my heaters are off.

Ok, I’ll start then, I am in Brittany 🌦️ 

> If you look to this video (and speak french), you can see that you can associate your heater to a temperature sensor : https://www.youtube.com/watch?v=gXSuO8J4Tvk (~2mn35) I thinks it's for calibration.

It’s been a long time since I tested Enki stuff (they gave me the box when I bought the heaters), it evolved nicely.
I am not sure the sensor associated to the heater is a direct link between them, I think it’s an Enki feature as you can do with any automation or some integration in Home Assistant. It means it would only work if the brain is online (HA or Enki).

**Reply by llaumgui** [reply]
> It’s been a long time since I tested Enki stuff (they gave me the box when I bought the heaters), it evolved nicely.
I am not sure the sensor associated to the heater is a direct link between them, I think it’s an Enki feature as you can do with any automation or some integration in Home Assistant. It means it would only work if the brain is online (HA or Enki).

When you manualy calibrate your sensor you see in the bottom a gauge. For me it's that the -12.8 and +12.7° calibration. But When you calibrate your sensor it's from a displayed temperature... It's this temperature that I would like to find, the point of calibration. Because without this, you can calibrate without reference.

**Comment by cperriot** [comment]
Hello, 
Got an Acova Taffetas 2 today, and it has the ZigbeeModel : TAFFETAS2 D1.00P1.03Z1.00 so the fix done in august make my heater configured as a Percale 2 so I wasn't getting the occupancy. I had to modify the acova.ts to get it right.
I guess Acova made a mess in the factory and assembled Percale with Taffetas modules/firmware....

**Reply by molusk** [reply]
Hello,
That’s what I was afraid of...
As we discussed earlier, I should merge Percale and Taffetas models in the acova.js as soon as we are confident in the configuration and @llaumgui can confirm both have the same features.
We also should update the documentation, currently occupancy sensor is not documented for example.
